SCIInstitute/ShapeWorks
# DeepSSM early stopping

This pull request covers issue #807 and fixes some bugs.

Bugs:

- There is an error from PR #785 where when percent variability is specified instead of number of dims for PCA in augmentation, then we don't have access to the number of PCA dims for initializing a model in testing. I fixed this by having the data augmentation call return the number of dimensions used. 
- For KDE sampling in data aug, sigma needed to be divided by the number of dims.
- When getting the mean and standard deviation for PCA score whitening, needed to be vector not single value. 

To test the early stopping:

- Checkout DeepSSM branch 
- Rerun conda_installs or with the shapeworks environment activated run `pip install /ShapeWorks/Python/DeepSSMUtilsPackage/` and  `pip install /ShapeWorks/Python/DataAugmentationUtilsPackage/`
- From 'Examples/Python/' run: `python RunUseCase.py --use_case deep_ssm --tiny_test`
- Check to see that is saves best and final model
